Sources/GoodExtensions/UIKit/UIViewExtensions.swift

```diff
@@ -316,3 +334,110 @@ public extension GRActive where Base: UIView {
 
 }
 
+// MARK: - Dimming
+
+public extension GRActive where Base: UIView {
+
+    private final class DimView: UIView {
+
+        init(intensity alpha: CGFloat = 0.3) {
+            super.init(frame: .zero)
+
+            translatesAutoresizingMaskIntoConstraints = false
+            backgroundColor = .black.withAlphaComponent(alpha)
+        }
+
+        required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+
+        func changeIntensity(_ alpha: CGFloat) {
+            backgroundColor = .black.withAlphaComponent(alpha)
+        }
+
+    }
+
+    func dim(intensity alpha: CGFloat = 0.3, duration: CGFloat = 0.3) {
+        undim()
+        let dimView = DimView(intensity: 0)
+
+        addSubview(dimView)
+        NSLayoutConstraint.activate([
+            dimView.topAnchor.constraint(equalTo: base.topAnchor),
+            dimView.leadingAnchor.constraint(equalTo: base.leadingAnchor),
+            dimView.trailingAnchor.constraint(equalTo: base.trailingAnchor),
+            dimView.bottomAnchor.constraint(equalTo: base.bottomAnchor)
+        ])
+
+        UIView.animate(withDuration: duration) {
+            dimView.changeIntensity(alpha)
+        }
+    }
+
+    func undim(duration: CGFloat = 0.3) {
+        let dimView = base.firstSubview(forType: DimView.self)
+        UIView.animate(
+            withDuration: duration,
+            animations: {
+                dimView?.changeIntensity(0)
+            },
+            completion: { [weak base] _ in
+                dimView?.removeFromSuperview()
+                base?.layoutSubviews()
```

Comment:
Je toto potrebne volat?

Reply:
Niesom si istym musim preskusat, chcel som len pridat co filip urobil na benzinole lebo sa mi ten dim a animacia velmi pacili ale neskusal som toto vyhadzovat. Teoreticky RemoveFromSuperview zavola automaticky layoutSubviews nad superview to by davalo zmysel ale aj keby nie tak sa to pri najblizsiom layoute odstrani a kedze je na 0.0 alpha ajtak ho uz nevidis takze suhlasim toto by sme mohli odstranit.

V metode bohuzial nieje spomenute ci explicitne prevola layout https://developer.apple.com/documentation/uikit/uiview/1622421-removefromsuperview